mods/default/nodes.lua

```diff
@@ -530,9 +530,12 @@ minetest.register_node("default:snow", {
 	}),
 
 	on_construct = function(pos)
-		pos.y = pos.y - 1
-		if minetest.get_node(pos).name == "default:dirt_with_grass" then
-			minetest.set_node(pos, {name = "default:dirt_with_snow"})
+		local belowPos = vector.new(pos)
+		belowPos.y = belowPos.y -1
+		if minetest.get_node(belowPos).name == "default:dirt_with_grass" then
+			minetest.set_node(belowPos, {name = "default:dirt_with_snow"})
+		elseif minetest.get_node(belowPos).name == "default:snow"  then
+			minetest.spawn_falling_node(pos)
```

Comment:
`minetest.get_node()` is intensive, so store `minetest.get_node(belowPos).name` in a variable, then use that variable twice.